XMonad/Layout/IndependentScreens.hs

```diff
@@ -101,13 +112,27 @@ unmarshallW = snd . unmarshall
 workspaces' :: XConfig l -> [VirtualWorkspace]
 workspaces' = nub . map unmarshallW . workspaces
 
+-- | wsConvertToNum takes a workspace and returns its ScreenId
+wsConvertToNum :: (Num b)=>W.Workspace [Char] l a-> b
+wsConvertToNum j = fromIntegral (toInteger $ digitToInt $ head $ takeWhile (/= '_') (W.tag j))
+
+-- | Workspace names are independently specified with each monitor.
+-- You can define your workspaces by calling withScreen for each screen:
+--
+-- > myConfig = def { workspaces = withScreen 0 ["web", "email", "irc"]  ++ withScreen 1 ["1", "2", "3"] }
```

Comment:
Can you share some idea as to how to implement it in a better way? My goal is to have it work like independent workspaces in Dwm.

Reply:
It's been a while since I used dwm. But if my memory serves, what you're describing in this documentation isn't anything like what dwm does. What dwm does is it has a fixed set of workspace names, and each screen gets their own independent copy of that name. The collection doesn't vary from screen to screen. That is what `IndependentScreens` is trying to duplicate.

If all the workspace names are different on each screen, then there's almost nothing to implement. You just have `workspaces = ["web", "email", "irc", "1", "2", "3"]` and then write your workspace-switching keybindings to only put `web` on the left and `1` on the right. You don't need any complicated marshalling and unmarshalling in sorts and displays and other mechanisms, because the fact that `web` only appears on the left is completely invisible to every other part of xmonad than the keybindings.

Reply:
> The collection doesn't vary from screen to screen. That is what `IndependentScreens` is trying to duplicate.
I see. So I think I will delete that line of documentation.